### PR TITLE
More cleanup to go with x25519 vectors

### PR DESCRIPTION
**Pull request by chris-wood**
This change cleans up some more stuff in the implementation and spec, fixing some errors along the way.

### CODE REVIEW

File: draft-irtf-cfrg-opaque.md [hunk 1]
```diff
@@ -1406,68 +1402,62 @@ following fields:
...
+  {{NISTCurves}}, after decoding `B` from its encoded input using
+  the compressed Octet-String-to-Elliptic-Curve-Point method according to {{NISTCurves}}.
+  The output is then encoded using the SerializeElement function of the OPRF
+  group described in {{OPRF, Section 2.1}}.
 
-#### Curve25519 Group
+#### 3DH Curve25519
 
-This section describes the implementation of the AKE group based on Curve25519,
+This section describes the implementation of the DH primitive based on Curve25519,
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:1453]: Same here

File: draft-irtf-cfrg-opaque.md [hunk 2]
```diff
@@ -1406,68 +1402,62 @@ following fields:
...
   key `k` based on seed (of length `Nseed = 32` bytes), as described in {{Section 5 of Curve25519}},
-  as well as the result of DiffieHellman(k, B), where B is the base point of Curve25519.
-- GenerateAuthKeyPair(): This is implemented by invoking DeriveAuthKeyPair
-  with `Nseed` random bytes as input.
-- SerializeElement(element): This is implemented using the similarly-named
-  function of the OPRF group described in {{OPRF, Section 2.1}}.
-- DiffieHellman(k, B): Implemented using the X25519 function in {{Section 5 of Curve25519}}.
+  as well as the result of DH(k, B), where B is the base point of Curve25519.
+- DH(k, B): Implemented using the X25519 function in {{Section 5 of Curve25519}}.
+  The output is then used raw, with no processing.
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:1460]: nit: used raw -> used as-is

File: draft-irtf-cfrg-opaque.md [hunk 3]
```diff
@@ -1406,68 +1402,62 @@ following fields:
...
-- SerializeElement(element): A function that maps `element` to a unique byte
-  array, mirrored from the definition of the similarly-named function of the
-  OPRF group described in {{OPRF, Section 2.1}}.
-- DiffieHellman(k, B): A function that performs the Diffie-Hellman operation
-  between the scalar input `k` and element `B`.
+- DeriveDHKeypair(seed): Derive a private and public DiffieHellman
+  key pair deterministically from the input `seed`. The type of the private key
+  depends on the implementation, whereas the type of the public key is a
+  byte string of `Npk` bytes.
+- DH(k, B): A function that performs the Diffie-Hellman operation
```
Comment (kevinlewi) [inline draft-irtf-cfrg-opaque.md:1414]: Why change the name here from "DiffieHellman(k, B)" to "DH(k, B)"? The latter feels less descriptive, and I am in favor of keeping it as DiffieHellman unless it conflicts with something else...

**Review by kevinlewi — CHANGES_REQUESTED** [review]
First round of requested changes, mainly looking to make sure that we keep parity with the name changes in the spec and the sage code

File: draft-irtf-cfrg-opaque.md [hunk 4]
```diff
@@ -1406,68 +1402,62 @@ following fields:
...
 
-#### ristretto255 Group
+#### 3DH ristretto255
 
-This section describes the implementation of the AKE group based on ristretto255,
+This section describes the implementation of the DH primitive based on ristretto255,
 as defined in {{?RISTRETTO=I-D.irtf-cfrg-ristretto255-decaf448}}.
 
-- DeriveAuthKeyPair(seed): This function is implemented as
+- DeriveDHKeypair(seed): This function is implemented as
```
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md:1426]: I had spelled it out fully first, but it was really quite long and I made it short. I don't feel strongly, though.

File: draft-irtf-cfrg-opaque.md [hunk 5]
```diff
@@ -1406,68 +1402,62 @@ following fields:
...
+  The output of this function is a unique, fixed-length byte string.
 
 Implementations for recommended groups in {{configurations}}, as well as groups
 covered by test vectors in {{test-vectors}}, are described in the following sections.
 
-#### ristretto255 Group
+#### 3DH ristretto255
 
-This section describes the implementation of the AKE group based on ristretto255,
+This section describes the implementation of the DH primitive based on ristretto255,
```
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md:1423]: It's just a name for a thing that has (1) a key derivation function and (2) a DH operation. We can call it whatever we want. Group feels wrong for that name.

File: draft-irtf-cfrg-opaque.md [hunk 6]
```diff
@@ -1875,7 +1867,7 @@ protocols such as TLS.
 
 ## Notable Design Differences
 
-[[RFC EDITOR: Please delete this section before publication.]]
+**xbefore publication.**
```
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md:1870]: Nope, reverting.

File: draft-irtf-cfrg-opaque.md (hunk 1 above)
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md]: Fixed.

File: draft-irtf-cfrg-opaque.md [hunk 7]
```diff
@@ -1406,68 +1402,62 @@ following fields:
...
+  {{Section 4 of RISTRETTO}} after decoding `B` from its encoded input using
+  the Decode function in {{Section 4.3.1 of RISTRETTO}}. The output is then
+  encoded using the SerializeElement function of the OPRF group described in
+  {{OPRF, Section 2.1}}.
 
-#### P-256 Group
+#### 3DH P-256
 
-This section describes the implementation of the AKE group based on NIST P-256,
+This section describes the implementation of the DH primitive based on NIST P-256,
```
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md]: Fixed.

File: draft-irtf-cfrg-opaque.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md]: Fixed.

File: draft-irtf-cfrg-opaque.md (hunk 3 above)
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md]: Reverted.

File: draft-irtf-cfrg-opaque.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md]: Fixed, and changed the domain separation tag in the function to match.

File: draft-irtf-cfrg-opaque.md [hunk 8]
```diff
@@ -1406,68 +1402,62 @@ following fields:
 {{ake-client}} and {{ake-server}} specify the inner workings of client and
 server functions, respectively.
 
-### AKE Group {#key-creation}
+### 3DH Primitives {#key-creation}
 
-We assume the following functions to exist for all candidate groups in this
+We assume the following functions to exist for all Diffie Hellman groups in this
```
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md]: Fixed.

File: draft-irtf-cfrg-opaque.md [hunk 9]
```diff
@@ -1406,68 +1402,62 @@ following fields:
 {{ake-client}} and {{ake-server}} specify the inner workings of client and
 server functions, respectively.
 
-### AKE Group {#key-creation}
+### 3DH Primitives {#key-creation}
```
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md]: Sure, done.

File: draft-irtf-cfrg-opaque.md [hunk 10]
```diff
@@ -1385,7 +1381,7 @@ def RecoverCredentials(password, blind, response,
   return (client_private_key, cleartext_credentials, server_public_key, export_key)
 ~~~
 
-## AKE Protocol {#ake-protocol}
+## 3DH Protocol {#ake-protocol}
```
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md]: Done.

File: draft-irtf-cfrg-opaque.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md]: Done.

**Review by kevinlewi — APPROVED** [review]
Excellent, this looks good to me! I still haven't verified that test vector changes just yet, but I can do that in a follow-up and report back with any issues. Let's go ahead and land this. 

Thank you for the quick turnaround!

**Pull request merged**